scripts/run_extension_tests.bat

```diff
@@ -3,12 +3,30 @@ setlocal
 
```

Comment:
I am unable to make the Linux tests to fail. A malformed CMakeLists.txt still allows the tests to pass.

Reply:
@vector-of-bool I think that is because the build/configure tests don't get executed on Linux at the moment: https://github.com/vector-of-bool/vscode-cmake-tools/pull/338/files#diff-a73238827bcc789254199f160d6f5d9aR17

Reply:
Oof. I missed that. Still, I'd like to see them running. @KoeMai, is there a reason they are disabled on Linux?

Reply:
I disabled linux and mac os. I first wanted feedback about the test way.
If you want to see the benefit of this way then look into 'add_prefered_generator_tests'. 

My open problem is the way to switch machine dependencies like vs or clang.. Or the make system. If you look on appveyor then you see it use only visual studio 11. 
My first idea for the different compiler support is on add_prefered_generator_tests.

Additional this PR contains the basic infrastructur. It is ready to use.

Reply:
Let's get this merged. I branched from this branch and I have some changes to do some Linux and macOS tests.